scripts/esb_sniffer/main.py

```diff
@@ -0,0 +1,226 @@
+        self.stop_evt = Event()
+        self.exit_exception = None
+
+    def start(self):
+        self.thread.start()
+
+    def stop(self):
+        self.stop_evt.set()
+
+class PipeData(Process):
+    # Read packets from device and write into extcap pipe
+    def __init__(self, event, dev):
+        super().__init__("PipeData", self.loop)
+        self.dev = dev
+        self.run = event
+
+    def loop(self):
+        self.pipe = open(PIPE_DATA, 'wb')
+        try:
+            while not self.stop_evt.is_set():
+                if self.run.is_set():
+                    pkt = self.dev.read()
+
+                    if pkt:
+                        try:
+                            self.pipe.write(pkt)
+                            self.pipe.flush()
+                        except BrokenPipeError:
+                            if not os.path.exists(PIPE_DATA):
+                                # removed pipe, cannot safetly continue
+                                raise BrokenPipeError
+                            else:
+                                continue
+                else:
+                    self.run.wait(timeout=30)
+        except Exception as e:
+            self.exit_exception = type(e).__name__
+            return 0
+
+        return 0
+
+    def __del__(self):
+        try:
+            self.pipe.close()
+        except Exception:
+            pass
+
+class PipeComm(Process):
+    # Start and stop device in response from extcap
+    def __init__(self, data_event, dev):
+        super().__init__("PipeComm", self.loop)
+        self.data_evt = data_event
+        self.dev = dev
+
+    def loop(self):
+        try:
+            self.pipe = open(PIPE_COMM, 'r')
+
+            while not self.stop_evt.is_set():
+                comm = self.pipe.read(len(COMM_START_CAPTURE))
+                if comm == COMM_START_CAPTURE:
+                    if self.dev.start() == 0:
+                        self.data_evt.set()
+                elif comm == COMM_STOP_CAPTURE:
+                    self.data_evt.clear()
+                    if self.dev.stop() != 0:
+                        pass
+                else:
+                    sleep(1)
+
+        except Exception as e:
+            self.data_evt.clear()
+            self.exit_exception = type(e).__name__
+            return 0
+
+        return 0
+
+    def __del__(self):
+        self.data_evt.clear()
+        try:
+            self.pipe.close()
+        except Exception:
+            pass
+
+class Watchdog(Process):
+    def __init__(self, tick=0.5):
+        super().__init__("Watchdog", self.loop)
+        self.tick = tick
+        self.watching_threads = []
+
+    def add_thread(self, t):
+        self.watching_threads.append(t)
+
+    def stop_all(self):
+        if not self.thread.is_alive():
+            return -1
+
+        for i in self.watching_threads:
+            if i.thread.is_alive():
+                self.logger.debug(f"Thread: {i.name} stopped")
```

Comment:
```suggestion
                self.logger.debug("Thread: %s stopped", i.name)
```
Also in other places.